### PR TITLE
Fix Show instance

### PR DESCRIPTION
**Pull request by pepeiborra**

### CODE REVIEW

File: ghcide/src/Development/IDE/Core/RuleTypes.hs [hunk 1]
```diff
@@ -421,7 +421,10 @@ newtype GhcSessionDeps = GhcSessionDeps_
         -- Required for interactive evaluation, but leads to more cache invalidations
         fullModSummary :: Bool
     }
-    deriving newtype (Eq, Show, Typeable, Hashable, NFData)
+    deriving newtype (Eq, Typeable, Hashable, NFData)
+
+instance Show GhcSessionDeps where
```
Comment (michaelpj) [inline ghcide/src/Development/IDE/Core/RuleTypes.hs:426]: What's wrong with the derived one? Worth a comment.

Comment (pepeiborra) [reply ghcide/src/Development/IDE/Core/RuleTypes.hs:426]: It prints True or False instead of the key name

Comment (pepeiborra) [reply ghcide/src/Development/IDE/Core/RuleTypes.hs:426]: The problem is that the deriving strategy is new type instead of stock

Comment (michaelpj) [reply ghcide/src/Development/IDE/Core/RuleTypes.hs:426]: So why not use the stock deriving?

Comment (pepeiborra) [reply ghcide/src/Development/IDE/Core/RuleTypes.hs:426]: Oh the stock instance is very verbose and bloats traces. I have changed the custom instance to not ignore the boolean for good measure.

**Review by wz1000 — APPROVED** [review]

**Pull request merged**